runtime/src/main/scala/akka/grpc/GrpcClientSettings.scala

```diff
@@ -4,33 +4,57 @@
 
 package akka.grpc
 
-import scala.concurrent.duration.Duration
+import akka.discovery.SimpleServiceDiscovery
+import akka.discovery.SimpleServiceDiscovery.{ Resolved, ResolvedTarget }
+import akka.grpc.internal.HardcodedServiceDiscovery
 
+import scala.concurrent.duration.{ Duration, FiniteDuration }
+import scala.collection.immutable
 import io.grpc.CallCredentials
+import scala.concurrent.duration._
 
 object GrpcClientSettings {
+  val DefaultResolveTimeout = 5.seconds
+  val DefaultPort = 443
+
   /** Scala API */
   def apply(host: String, port: Int): GrpcClientSettings =
-    new GrpcClientSettings(host, port)
+    new GrpcClientSettings(host, hardcodedServiceDiscovery(host, port))
+
+  /** Scala API */
+  def apply(name: String, serviceDiscovery: SimpleServiceDiscovery) =
+    new GrpcClientSettings(name, serviceDiscovery)
 
   /** Java API */
   def create(host: String, port: Int): GrpcClientSettings = apply(host, port)
+
+  /** Java API */
+  def create(name: String, serviceDiscovery: SimpleServiceDiscovery) =
+    new GrpcClientSettings(name, serviceDiscovery)
+
+  private def hardcodedServiceDiscovery(host: String, port: Int) = new HardcodedServiceDiscovery(Resolved(host, immutable.Seq(ResolvedTarget(host, Some(port)))))
 }
 
 final class GrpcClientSettings private (
-  val host: String,
-  val port: Int,
+  val name: String,
+  val serviceDiscovery: SimpleServiceDiscovery,
+  val defaultPort: Int = 443,
```

Comment:
Perhaps the default port should depend on whether TLS is enabled? (we currently determine that by checking if `trustedCaCertificate` is defined, might be good to make that explicit by introducing a 'def useTls() = trustedCaCertificate.isDefined`, this might change based on the work by @richdougherty for #89)